ParticleObjects/Source/materialSource_class.f90

```diff
@@ -0,0 +1,234 @@
+module materialSource_class
+
+  use numPrecision
+  use universalVariables,      only : OUTSIDE_MAT, VOID_MAT
+  use genericProcedures,       only : fatalError, rotateVector
+  use dictionary_class,        only : dictionary
+  use RNG_class,               only : RNG
+  use charMap_class,           only : charMap
+
+  use particle_class,          only : particleState, P_NEUTRON
+  use source_inter,            only : source, kill_super => kill
+
+  use geometry_inter,          only : geometry
+  use neutronMaterial_inter,   only : neutronMaterial, neutronMaterial_CptrCast
+  use nuclearDataReg_mod,      only : ndReg_getNeutronCE => getNeutronCE, &
+                                      ndReg_getNeutronMG => getNeutronMG
+  use nuclearDatabase_inter,   only : nuclearDatabase
+  use ceNeutronDatabase_inter, only : ceNeutronDatabase
+  use mgNeutronDatabase_inter, only : mgNeutronDatabase
+
+  implicit none
+  private
+
+  !!
+  !! Neutron Source from a given material
+  !!
+  !! Places neutrons uniformly in regions with given material.
+  !! Angular distribution is isotropic.
+  !!
+  !! Can be fed a bounding box to increase sampling efficiency
+  !!
+  !! Private members:
+  !!   isMG   -> is the source multi-group? (default = .false.)
+  !!   bottom -> Bottom corner (x_min, y_min, z_min)
+  !!   top    -> Top corner (x_max, y_max, z_max)
+  !!   E      -> Source site energy [MeV] (default = 1.0E-6)
+  !!   G      -> Source site Group (default = 1)
+  !!
+  !! Interface:
+  !!   source_inter Interface
+  !!
+  !! Sample Dictionary Input:
+  !!   matsource {
+  !!     type materialSource;
+  !!     #data MG; #
+  !!     #E 15.0;  #
+  !!     #G 7;     #
+  !!     #bb (-x -y -z +x +y +z); #
```

Comment:
I would call this boundingBox, it's longer but more descriptive.

```suggestion
  !!     #boundingBox (-x -y -z +x +y +z); #
```

Reply:
I thought this was more consistent with terminology we had elsewhere, although perhaps I misremembered?

Reply:
I don't remember having a bounding box specified in the input file anywhere else honestly.l Surely it's not in the manual as an option!